src/src/content/work/thesis.md

```diff
@@ -0,0 +1,37 @@
+
+The practical aspect required us to design and implement a solution, either software or hardware-based, to tackle a real-world problem. My teammate, Clemens Schlipfinger, and I decided to focus exclusively on software to ensure reliability. Hardware can introduce unpredictable issues, but software—especially when deployed in robust Docker containers—offered a consistent environment that worked seamlessly everywhere.
+
+What made our project particularly exciting was our partnership with Siemens AG, a renowned multinational company specializing in energy, automation, and digitalization. This collaboration was rare among our peers and added a professional dimension to our work. Siemens tasked us with developing a microservice-based software solution for analyzing anomalies in Europe's power grid.
+
+The system we built comprised several components, each designed to handle a specific function. At the core was a Kafka application responsible for ingesting data—power grid anomaly reports we termed Findings. These reports detailed failures and irregularities across Europe’s intricate power network, comprising millions of interconnected components like transistors, switches, and generators. Kafka transmitted the most recent anomalies to a PostgreSQL database, which was then made accessible via a GraphQL API. This API provided comprehensive features like pagination, filtering, and querying to ensure seamless integration with our Angular-based frontend.
+
+The frontend was designed as a user-friendly dashboard with three key sections:
+
+1. Anomalies Table – A searchable, paginated, and filterable table for quick navigation of anomalies.
+
+2. Graph View – A dynamic visualisation of anomaly data, making it easier to pinpoint the source of an issue within the vast, complex grid.
+
+3. Overall Dashboard – Key metrics and insights for quick, actionable overviews.
+
+Clemens and I divided our responsibilities evenly. Clemens handled the backend systems, including Kafka, PostgreSQL, and the GraphQL API, ensuring a reliable and efficient data pipeline. I took on the role of project leader, coordinating communication with Siemens while developing the frontend and ensuring the seamless integration of the various components. This collaboration allowed us to play to our strengths and maintain a balanced workload.
+
+Our final product not only met Siemens' requirements but also demonstrated the power of a well-designed microservice architecture in solving complex, real-world problems. It was immensely rewarding to see our software functioning reliably and providing actionable insights to tackle power grid anomalies.
```

Comment:
wir haben ja eigendlich keine microservice architecture gemacht ...  aber egal

Reply:
Thank you very much, I edited it so it's a SOA now like you suggested 🎉